runtime/src/accounts_db.rs

```diff
@@ -2466,21 +2466,32 @@ impl AccountsDb {
     }
 
     pub fn shrink_all_slots(&self, is_startup: bool) {
+        const MAX_DIRTY_STORES: usize = 10_000;
         if is_startup && self.caching_enabled {
             let slots = self.all_slots_in_storage();
-            let chunk_size = std::cmp::max(slots.len() / 8, 1); // approximately 400k slots in a snapshot
-            slots.par_chunks(chunk_size).for_each(|slots| {
-                for slot in slots {
-                    self.shrink_slot_forced(*slot, is_startup);
-                }
-            });
+            let outer_chunk_size = 2000;
+            let chunk_size = std::cmp::max(outer_chunk_size / 8, 1);
+            slots.chunks(outer_chunk_size).for_each(|chunk|
+                {
+                    chunk.par_chunks(chunk_size).for_each(|slots| {
+                        for slot in slots {
+                            self.shrink_slot_forced(*slot, is_startup);
+                        }
+                    });
+                    if self.dirty_stores.len() > MAX_DIRTY_STORES {
+                        self.clean_accounts(None, false);
```

Comment:
should this pass `is_startup` instead of `false`?, i.e. `self.clean_accounts(None, is_startup)`

Reply:
ah, yes.